helm/tenant/templates/tenant.yaml

```diff
@@ -1,35 +1,35 @@
-{{- with .Values.tenant }}
+{{- with .Values }}
 apiVersion: minio.min.io/v2
 kind: Tenant
 metadata:
-  name: {{ .name }}
+  name: {{ .tenant.name }}
   ## Optionally pass labels to be applied to the statefulset pods
   labels:
     app: minio
-  {{- if dig "metrics" "enabled" false . }}
+  {{- if dig "metrics" "enabled" false .tenant }}
   ## Annotations for MinIO Tenant Pods
   annotations:
     prometheus.io/path: /minio/v2/metrics/cluster
-    prometheus.io/port: {{ dig "metrics" "port" 9000 . | quote }}
+    prometheus.io/port: {{ dig "metrics" "port" 9000 .tenant | quote }}
     prometheus.io/scrape: "true"
-    prometheus.io/scheme: {{ dig "metrics" "protocol" "http" . | quote }}
+    prometheus.io/scheme: {{ dig "metrics" "protocol" "http" .tenant | quote }}
   {{- end }}
-{{- if dig "scheduler" "name" "" . }}
+{{- if dig "scheduler" "name" "" .tenant }}
 scheduler:
-  name: {{ dig "scheduler" "name" "" . }}
+  name: {{ dig "scheduler" "name" "" .tenant }}
 {{- end }}
 spec:
-  image: "{{ .image.repository }}:{{ .image.digest | default .image.tag }}"
-  imagePullPolicy: {{ dig "image" "pullPolicy" "IfNotPresent" . }}
-  {{- if dig "imagePullSecret" "name" "" . }}
+  image: "{{ .tenant.image.repository }}:{{ .tenant.image.digest | default .tenant.image.tag }}"
+  imagePullPolicy: {{ dig "image" "pullPolicy" "IfNotPresent" .tenant }}
+  {{- if dig "imagePullSecret" "name" "" .tenant }}
   imagePullSecret:
-    name: {{ dig "imagePullSecret" "name" "" . }}
+    name: {{ dig "imagePullSecret" "name" "" .tenant }}
   {{- end }}
   ## Secret with default environment variable configurations
   configuration:
-    name: {{ .configuration.name }}
+    name: {{ .secrets.existingSecret | default .secrets.name }}
```

Comment:
I can't get this from your pr.

Reply:
I am not entirely sure I understand you correctly. What I did here is removing configuration from values.yml while leaving it in tenant.yml. Since configuration.name from values.yaml is the same secret as secret.name or secret.exsistingSecret, I reference the configuration.name to that in tenant.yaml.

If you mean something else, please let me know

Reply:
Yeah. I can't find out the `tenant.secrets.existingSecret` field at the values.yaml. What did I miss?

Reply:
Thats the beauty of this PR if I say so myself. The tenant references the `secrets` from values as shown below. Hence all the .tenant in tenant.yaml and as base {{- with .Values }}.
```yaml
secrets:
  name: myminio-env-configuration
  # MinIO root user and password
  accessKey: minio
  secretKey: minio123
  ## Set the value for existingSecret to use a pre created secret and dont create default one
  # existingSecret: random-env-configuration
## MinIO Tenant Definition
```

This removes the need of specifying twice the secret in values.yaml.